cdap-common/src/main/java/co/cask/cdap/common/HttpExceptionHandler.java

```diff
@@ -54,9 +54,10 @@ public void handle(Throwable t, HttpRequest request, HttpResponder responder) {
       responder.sendStatus(HttpResponseStatus.METHOD_NOT_ALLOWED);
     } else if (t instanceof UnauthorizedException) {
       logWithTrace(request, t);
-      HttpResponseStatus status = ((UnauthorizedException) t).isAuthenticationError() ?
-        HttpResponseStatus.UNAUTHORIZED : HttpResponseStatus.FORBIDDEN;
-      responder.sendStatus(status);
+      responder.sendStatus(HttpResponseStatus.UNAUTHORIZED);
+    } else if (t instanceof co.cask.cdap.security.spi.authorization.UnauthorizedException) {
```

Comment:
Yeah, basically the current ``UnauthorizedException`` is really an ``AuthenticationException``. However, I realized that this may have to stay for a couple of releases, since the current ``UnauthorizedException`` (and other exceptions from ``cdap-common``) are used in the method signatures of ``cdap-clients``, and should stay backwards-compatible. We may have to live with this for a couple of releases before removing it from ``cdap-common``.